radsat-sk/operation/subsystems/RAntenna.c

```diff
@@ -0,0 +1,225 @@
+/**
+ * @file RAntenna.c
+ * @date Jan 15, 2022
+ * @author Addi Amaya (caa746)
+ */
+
+#include <RAntenna.h>
+#include <satellite-subsystems/IsisAntS.h>
+#include <string.h>
+#include <hal/errors.h>
+
+/***************************************************************************************************
+                                  PRIVATE DEFINITIONS AND VARIABLES
+***************************************************************************************************/
+
+/** Struct for defining Antenna Systems I2C Addresses. Configured as a single bus, both address write to same location */
+ISISantsI2Caddress RAntennaI2Caddress = {ANTENNA_I2C_SLAVE_ADDR_PRIMARY,ANTENNA_I2C_SLAVE_ADDR_REDUNANT};
+
+/** Struct that holds the current deployment status of the antenna*/
+antennaDeploymentStatus RAntennaStatus;
+
+/** Side A of the Antenna system*/
+ISISantsSide RISISASide = isisants_sideA;
+
+/** Side B of the Antenna system*/
+ISISantsSide RISISBSide = isisants_sideB;
+
+/** Armed Status for the antenna system*/
+ISISantsArmStatus RArmed = isisants_arm;
+
+/** Disarmed status for the antenna system*/
+ISISantsArmStatus RDisarmed = isisants_disarm;
+
+/** Track whether the antenna has been initialized yet */
+static int antennaInitialized = 0;
+
+/** Track Number of times the antenna attempted to deploy*/
+static int antennaDeploymentAttempts = 0;
+
+
+/***************************************************************************************************
+                                             PUBLIC API
+***************************************************************************************************/
+
+/**
+ * Initializes the ISISpace Antenna driver
+ *
+ * @return 0 for success, non-zero for failure. See hal/errors.h for details.
+ */
+int antennaInit(void) {
+
+	// only allow initialization once (return without error if already initialized)
+	if (antennaInitialized)
+		return 0;
+
+	int error = IsisAntS_initialize(&RAntennaI2Caddress, ANTENNAS_ON_BOARD);
+
+	// update flag if successfully initialized
+	if (!error)
+		antennaInitialized = 1;
+
+	// TODO: record errors (if present) to System Manager
+
+	return error;
+}
+
+/**
+ * Activate the auto-deployment mechanism on ISISpace Antenna System which will attempt to sequentially
+ * deploy all anntennas present on the system without intervention. Has a built in timer per antenna deployment
+ * @return 0 for success, non-zero for failure. See hal/errors.h for details.
+ */
+int antennaDeploymentAttempt(void) {
+
+	//Initialize the union for status
+	ISISantsStatus RISISantsStatus = { .fields = { 0 } };
```

Comment:
- styleguide (variable name, leave a space between slashes and the words in a comment)